File: UI/wwwroot/Facturacion/FrontModel/Detalle_Compra.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Producto }  from './Cat_Producto.js'
import { Tbl_Compra }  from './Tbl_Compra.js'
import { Tbl_Lotes }  from './Tbl_Lotes.js'
import {WAjaxTools} from "../../WDevCore/WModules/WAjaxTools";
class Detalle_Compra extends EntityClass {
   constructor(props) {
       super(props, 'EntityFacturacion');
       for (const prop in props) {
           this[prop] = props[prop];
       };
       this.Aplica_Iva = undefined;
       this.Precio_Venta = undefined;
   }
   /**@type {Number}*/ Id_Detalle_Compra;
   /**@type {Number}*/ Cantidad;
   /**@type {Number}*/ Precio_Unitario;
   /**@type {Number}*/ SubTotal;
   /**@type {Number}*/ Iva;
   /**@type {Number}*/ Total;
   /**@type {String}*/ Presentacion;   
   /**@type {Object}*/  Datos_Producto_Lote;
   /**@type {Cat_Producto} ManyToOne*/ Cat_Producto;
   /**@type {Tbl_Compra} ManyToOne*/ Tbl_Compra;
   /**@type {Array<Tbl_Lotes>} OneToMany*/ Tbl_Lotes;
}
export { Detalle_Compra }

File: UI/CAPA_NEGOCIO/Facturacion/Mapping/Detalle_Compra.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   

    public class Detalle_Compra : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? Id_Detalle_Compra { get; set; }
        public int? Id_Compra { get; set; }
        public int? Id_Producto { get; set; }
        public Double? Cantidad { get; set; }
        public Double? Precio_Unitario { get; set; }
        public Double? Precio_Venta { get; set; }
        public Double? SubTotal { get; set; }
        public Double? Iva { get; set; }
        public Double? Total { get; set; }
        public string? Presentacion { get; set; }
        //Datos del producto que se agregara al lote
        public object? Datos_Producto_Lote { get; set; }
        
        [ManyToOne(TableName = "Cat_Producto", KeyColumn = "Id_Producto", ForeignKeyColumn = "Id_Producto")]
        public Cat_Producto? Cat_Producto { get; set; }
        [ManyToOne(TableName = "Tbl_Compra", KeyColumn = "Id_Compra", ForeignKeyColumn = "Id_Compra")]
        public Tbl_Compra? Tbl_Compra { get; set; }
        [OneToMany(TableName = "Tbl_Lotes", KeyColumn = "Id_Detalle_Compra", ForeignKeyColumn = "Id_Detalle_Compra")]
        public List<Tbl_Lotes>? lotes { get; set; }
    }
}
